### ISSUE TITLE
my code is replaced with ✂prettier:content✂

### ISSUE THREAD

**Issue by Mlocik97** [sveltejs/prettier-plugin-svelte#240]
I still have this problem, with for example this kind of code:

```html
{@html `<style>
        body { background-color: ${bg_col}; }
    </style>`}
```

it's replaced for:

```html
{@html `<style ✂prettier:content✂="CgkJYm9keSB7IGJhY2tncm91bmQtY29sb3I6ICR7bWYuYmFja2dyb3VuZF9jb2xvcn07IH0KCQ=="></style>`}
```

tried even comment above {@html ...}, but still it was transforming it to that ugly prettier:content thing.

https://www.youtube.com/watch?v=lHK66Y06OoE

@dummdidumm

_Originally posted by @Mlocik97 in https://github.com/sveltejs/prettier-plugin-svelte/issues/59#issuecomment-890403216_

**Comment by Mlocik97** [comment]
so https://github.com/sveltejs/prettier-plugin-svelte/pull/130 didn't fix it.

**Comment by navorite** [comment]
I have the same issue as well when using this code: `<iframe
    class="content"
    id="inlineviewer"
    title="Inline Viewer"
    name="Inline Viewer"
    srcdoc={`<!DOCTYPE html>
<html lang='en'>
  <head>
    <meta charset='utf-8'/>
    <title>Inline Viewer</title>
    <style></style>
  </head>
  <body>
  </body>
</html>`}
/>`

It produces: 
`<iframe
    class="content"
    id="inlineviewer"
    title="Inline Viewer"
    name="Inline Viewer"
    srcdoc={`<!DOCTYPE html>
<html lang='en'>
  <head>
    <meta charset='utf-8'/>
    <title>Inline Viewer</title>
    <style ✂prettier:content✂=""></style>
  </head>
  <body>
  </body>
</html>`}
/>`

**Comment by codingwithchris** [comment]
@dummdidumm This issue still exists as of:

```
"prettier": "^2.6.2",
"prettier-plugin-svelte": "^2.7.0",
```

**Example:**

Input
```
import { getCssText } from '@tokens';
<svelte:head>
	{@html `<style>${getCssText()}</style>`}
</svelte:head>
```

Output
```
import { getCssText } from '@tokens';
<svelte:head>
    {@html `<style ✂prettier:content✂="JHtnZXRDc3NUZXh0KCl9"></style>`}
</svelte:head>
```

**Comment by bjornarhagen** [comment]
Using version 2.8.0 I'm getting this:

Original:
```svelte
{'<script></script>'}
{`<script></script>`}
```

Formatted:
```svelte
{'<script ✂prettier:content✂="">{}</script>'}
{`<script ✂prettier:content✂="">{}</script>`}
```

**Comment by justingolden21** [comment]
My code:
```
{@html `<script type="application/ld+json">${JSON.stringify(({
	"@context": "https://schema.org",
    "@type": "WebSite",
    "name": "Desktop Clock"
    "url": '"'+appURL+'"',
}))}
```
is replaced with:
```
{@html `<script type="application/ld+json" ✂prettier:content✂="JHtKU09OLnN0cmluZ2lmeSgoewoJIkBjb250ZXh0IjogImh0dHBzOi8vc2NoZW1hLm9yZyIsCiAgICAiQHR5cGUiOiAiV2ViU2l0ZSIsCiAgICAibmFtZSI6ICJEZXNrdG9wIENsb2NrIgogICAgInVybCI6ICciJythcHBVUkwrJyInLAp9KSl9Cgk=">{}</script>`}
```

**Comment by justingolden21** [comment]
Happening in July 2023... 

Simple reproducible example in a svelte file conditionally rendering SVGs:

```svelte
<script>
    export let name;
</script>
{#if name === 'sunlight'}
	<!-- prettier-ignore -->
	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 423.9 590.2">
        <circle class="fill-accent-500" cx="65.8" cy="65.8" r="65.8" />
   </svg>
{:else if name === 'time'}
	<!-- prettier-ignore -->
	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 675.67 631.95">
        <style>
            .st0{fill:#94A3B8;} /*400*/
        </style>
            <circle cx="166" cy="508.87" r="17"/>
        </svg>
{/if}
```

Is replaced with:
```
<script>
	export let name;
</script>

{#if name === 'sunlight'}
	<!-- prettier-ignore -->
	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 423.9 590.2">
        <circle class="fill-accent-500" cx="65.8" cy="65.8" r="65.8" />
   </svg>
{:else if name === 'time'}
	<!-- prettier-ignore -->
	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 675.67 631.95">
        <style ✂prettier:content✂="CiAgICAgICAgICAgIC5zdDB7ZmlsbDojOTRBM0I4O30gLyo0MDAqLwogICAgICAgIA=="></style>
            <circle cx="166" cy="508.87" r="17"/>
        </svg>
{/if}
```

**Comment by digvijayad** [comment]
For me the issue because of PhpStorms [ `Reformat code` or `Optimize Imports`](https://github.com/sveltejs/prettier-plugin-svelte/issues/70#issuecomment-1636920670) options.

I disabled those from my git commit checks and am using either the PhpStorm Prettier plugin's `Run-on-save` option or `Ctrl+Shift+Alt+P - Reformat with Prettier`

**Comment by huangyinhaow** [comment]
Can confirm, this is still happening.

**Comment by justingolden21** [comment]
Why is this issue closed? Do we need to make a new issue to reopen it?

**Comment by dummdidumm** [comment]
Yes, if people still have certain code snippets that fail using the latest version of this plugin, then they should open a new issue with that code snippet.

**Comment by benediktms** [comment]
I also had this issue in webstorm, but similar to @digvijayad I got around it by disabling the build in `Reformat code` option and relying on prettier. I think it's probably an issue with both formatters trying to run at the same time. Running only prettier on save solved the issue for me. I think this is webstorm/jetbrains specific, at least in my case, because formatting the same file in VSCode worked fine.

**Comment by imagoiq** [comment]
I've just get the same issue with Intellij, but in my case disabling `Optimize imports` for HTML seems to solve it.

**Comment by DannyFestor** [comment]
> I've just get the same issue with Intellij, but in my case disabling `Optimize imports` for HTML seems to solve it.

Thank you, this fixed it for me in PHPstorm.

**Comment by MS90D3V** [comment]
I have Optimize imports disabled and it still happens to me for IntelliJ.

![Image](https://github.com/user-attachments/assets/99ef8560-66ca-49bc-8fe6-f7d55f491bb8)

Only temp fix I found is to run formatting from CLI and then it works again (for a while) in IntelliJ with formatting. Befofore it breaks again.

**Comment by rudolfbyker** [comment]
The only way I found to work around this is to disable Prettier entirely in the IDE (IntelliJ IDEA Ultimate 2025.1) and try to remember to run it from the command line.
